### ISSUE TITLE
Could not resolve "croner" - Error while bundling for production

### ISSUE THREAD

**Issue by rdzidziguri** [kibertoad/toad-scheduler#163]
While trying to build production, the following error is being thrown that prevents from building the application. This has never been an issue and has always worked with the `2.0.0` version
```
tsc --noEmit && node ./build/esbuild.js --metafile=meta.json
X [ERROR] Could not resolve "croner"
    ../../node_modules/toad-scheduler/dist/lib/engines/cron/CronJob.js:24:31:
      24 │         const croner = require('croner');
```

**Comment by kibertoad** [comment]
Thank you for reporting this. Need to think how to approach this, I was hoping that lazy-loading it would help keep the dependency optional, but looks like bundlers are resolving it eagerly anyway.

**Comment by rdzidziguri** [comment]
Meanwhile, are you rolling out another version that reverts this, or do you recommend we downgrade to 2.0.0?

**Comment by kibertoad** [comment]
downgrading to 2.1.1 should suffice, but I'll work on this tomorrow

**Comment by Hexagon** [comment]
@kibertoad Just depend it 😄 

I won't add any third party dependencies - ever. It's small. And bundlers will tree-shake if needed 👍

**Comment by rdzidziguri** [comment]
@kibertoad, Any update regarding this one? Actually @Hexagon had a really good point ;)

**Comment by kibertoad** [comment]
yeah, had few ideas, will try them out soon, but worst case scenario just make croner a direct dependency

**Comment by kibertoad** [comment]
@rdzidziguri you can already include croner in your package.json and make error go away

**Comment by rdzidziguri** [comment]
Sure, that is the way, but we have strict rules regarding dependencies, mainly not including this way bundlers will. The good thing is that Croner has zero dependencies and is very tiny, so we are evaluating its effect on projects, especially in bundle sizing.

**Comment by kibertoad** [comment]
my concern exactly. will try finding the way to avoid including it

**Comment by kibertoad** [comment]
@rdzidziguri Sorry it took so long, but - going forward solution will be extracting cron-related logic into a separate optional package, as I don't think there is a nicer solution available.

How bad was the impact on the bundle size after including Croner? would you recommend just making it a runtime dependency?

**Comment by kibertoad** [comment]
Resolved as of v3.x — `croner` is now a regular `dependencies` entry (currently `^10.0.1`) instead of a lazy `require()`, so bundlers will resolve it normally. Closing. Thanks for the report!
